app_test_suite/steps/scenarios/upgrade.py

```diff
@@ -183,7 +209,17 @@ def run_tests(self, config: argparse.Namespace, context: Context) -> None:
             logger.debug(f"Deleting AppCatalog '{app_catalog_cr.name}'.")
             app_catalog_cr.delete()
 
-        # TODO: save upgrade metadata
+        # save metadata, if requested
+        if get_config_value_by_cmd_line_option(config, key_cfg_upgrade_save_metadata):
+            self._save_metadata(
+                app_name,
+                chart_version,
+                context[CONTEXT_KEY_CHART_YAML]["appVersion"],
+                context[CONTEXT_KEY_STABLE_CHART_YAML]["version"],
+                context[CONTEXT_KEY_STABLE_CHART_YAML]["appVersion"],
```

Comment:
Can we have a comment here explaining that we refer to appVersion as the upstream chart version?